anndata/base.py

```diff
@@ -600,11 +601,19 @@ class AnnData(IndexMixin, metaclass=utils.DeprecationMixinMeta):
     }
 
     def __init__(
-            self, X=None, obs=None, var=None, uns=None,
-            obsm=None, varm=None, raw=None,
-            dtype='float32', single_col=False,
-            filename=None, filemode=None,
-            asview=False, oidx=None, vidx=None):
+        self,
+        X: Optional[Union[np.ndarray, sparse.spmatrix]] = None,
+        obs: Optional[Union[pd.DataFrame, Mapping[Any, Iterable[Any]], np.ndarray]] = None,
+        var: Optional[Union[pd.DataFrame, Mapping[Any, Iterable[Any]], np.ndarray]] = None,
+        uns: Optional[Mapping[Any, Any]] = None,
+        obsm: Optional[np.ndarray] = None,
```

Comment:
since very recently - to enable interoperability with seurat - we also have dict init for obsm, varm and raw

Reply:
fixed